src/cli.rs

```diff
@@ -152,6 +152,23 @@ pub struct RootOpts {
     #[arg(short, long, env = "VECTOR_WATCH_CONFIG")]
     pub watch_config: bool,
 
+    /// Method for watching config
+    ///
+    /// By default `vector` use `inotify` which is recommended for linux based system
+    /// `poll` watcher can be used where inotify not work. ie, attaching config using NFS
+    #[arg(long, default_value = "inotify", env = "VECTOR_WATCH_METHOD")]
+    pub watch_config_method: WatchConfigMethod,
+
+    /// poll for changes in configuration file on given interval
+    ///
+    /// This config is only applicable if poll is enabled in `--watch-config-method`
+    #[arg(
+        long,
+        env = "VECTOR_WATCH_CONFIG_POLL_INTERVAL_SECONDS",
+        default_value = "30"
+    )]
+    pub watch_config_poll_interval_seconds: NonZeroU64,
```

Comment:
I think we should update this documentation to explain the polling watcher and when to use it

https://vector.dev/docs/administration/management/#automatic-reloading-on-configuration-change

But first lets see what maintainers think about documenting this

Reply:
where we need to add this in vector codebase?

Reply:
A mention here along with the Rustdocs (that you already have) should suffice: https://vector.dev/docs/administration/management/#automatic-reloading-on-configuration-change

Reply:
I see that you resolved this comment @amribm but this documentation was not updated. I would prefer to update that before merging this

Reply:
Added documentation